pkg/plugins/manager/manager_test.go

```diff
@@ -106,23 +131,66 @@ func TestPluginManager_Init(t *testing.T) {
 	})
 
 	t.Run("With external back-end plugin with valid v2 signature", func(t *testing.T) {
+		pluginsDir := "testdata/valid-v2-signature"
+		pluginFolder := pluginsDir + "/plugin"
 		pm := createManager(t, func(manager *PluginManager) {
-			manager.Cfg.PluginsPath = "testdata/valid-v2-signature"
+			manager.Cfg.PluginsPath = pluginsDir
 		})
 		err := pm.Init()
 		require.NoError(t, err)
 		require.Empty(t, pm.scanningErrors)
 
-		const pluginID = "test"
-		assert.NotNil(t, pm.plugins[pluginID])
-		assert.Equal(t, "datasource", pm.plugins[pluginID].Type)
-		assert.Equal(t, "Test", pm.plugins[pluginID].Name)
-		assert.Equal(t, pluginID, pm.plugins[pluginID].Id)
-		assert.Equal(t, "1.0.0", pm.plugins[pluginID].Info.Version)
-		assert.Equal(t, plugins.PluginSignatureValid, pm.plugins[pluginID].Signature)
-		assert.Equal(t, plugins.GrafanaType, pm.plugins[pluginID].SignatureType)
-		assert.Equal(t, "Grafana Labs", pm.plugins[pluginID].SignatureOrg)
-		assert.False(t, pm.plugins[pluginID].IsCorePlugin)
+		// capture manager plugin state
+		datasources := pm.dataSources
+		panels := pm.panels
+		apps := pm.apps
+
+		verifyPluginManagerState := func() {
+			// verify plugin manager has loaded core plugins successfully
+			assert.Empty(t, pm.scanningErrors)
+			assert.Len(t, pm.Plugins(), 45)
+			assert.Len(t, datasources, 21)
+			assert.Len(t, panels, 24)
+			assert.Len(t, apps, 0)
+
+			// verify plugin has been loaded successfully
+			const pluginID = "test"
+
+			assert.NotNil(t, pm.plugins[pluginID])
+			assert.Equal(t, "datasource", pm.plugins[pluginID].Type)
+			assert.Equal(t, "Test", pm.plugins[pluginID].Name)
+			assert.Equal(t, pluginID, pm.plugins[pluginID].Id)
+			assert.Equal(t, "1.0.0", pm.plugins[pluginID].Info.Version)
+			assert.Equal(t, plugins.PluginSignatureValid, pm.plugins[pluginID].Signature)
+			assert.Equal(t, plugins.GrafanaType, pm.plugins[pluginID].SignatureType)
+			assert.Equal(t, "Grafana Labs", pm.plugins[pluginID].SignatureOrg)
+			assert.Equal(t, pluginFolder, pm.plugins[pluginID].PluginDir)
+			assert.False(t, pm.plugins[pluginID].IsCorePlugin)
```

Comment:
I would recommend using go-cmp here, for a more comprehensive equality assertion.